### DISCUSSION TITLE
about Instrument2Spectrogram

### DISCUSSION THREAD

**Discussion by zdwangjian5** [groops-devs/groops#269] [Q&A] [answered]
I want to get the ASD (amplitude spectrum) of KBR/LRI1B 2019-11-01 'range', however when i use the Instrument2Spectrogram, the windowlength is hard to set. If I set to 1000, there are many NAN. when I set to 6hours, the output file is too large and can not be opened by notebook or other txt software.
There is another " Instrument2PowerSpectralDensity", but the output is somewhat different from the result by ltpda tool. How should I do ? Hoping for your help!

**Comment by akvas** [comment]
Hi @zdwangjian5,

Instrument2PowerSpectralDensity uses a slightly different algorithm than `ltpda.psd` because it can also handle irregularly spaced time series. However, for regularly spaced, stationary time series it yields the same results as MATLAB's `pwelch` which ltpda.psd  uses under the hood (if `noverlap=0`).

The workflow for a reliable PSD estimate with GROOPS looks like the following:
1. Optional: If transient signals are expected (e.g., trends, jumps, ...) use InstrumentDetrend to remove any non-stationary signal components. For SST range measurements this might be crucial because you have to consider potential phase breaks, i.e. a change in the unknown cycles resulting in a sudden jump of the range measurements. This will certainly lead to hard to interpret PSD estimates.

2. Use InstrumentSynchronize to divide the instrument file into arcs with the desired window length
3. Use Instrument2PowerSpectralDensity to estimate the PSD

Here are a few sanity checks to determine why there might be differences between GROOPS and ltpda:

- compare the frequency vectors of both functions to check whether the frequency spacing is the same
- GROOPS only supports a rectangular window, ltpda might use a different one per default
- GROOPS does not use overlapping segments in the PSD estimate, while MATLAB's  `pwelch` probably uses some amount of overlap. If possible pass `noverlap=0`to ltpda to get the same behavior

Cheers,
Andreas

**Reply by zdwangjian5** [reply, marked as answer]
Thank you for your help!
